shared/api/storage.go

```diff
@@ -0,0 +1,37 @@
+package api
+
+// StoragePoolConfig represents the fields of a LXD storage pool.
+type StoragePool struct {
+	PoolName       string   `json:"name" yaml:"name"`
+	PoolDriver     string   `json:"driver" yaml:"driver"`
+	PoolUsedBy     []string `json:"used_by" yaml:"used_by"`
+	StoragePoolPut `yaml:",inline"`
+}
+
+type StoragePoolPut struct {
+	PoolConfig map[string]string `json:"config" yaml:"config"`
+}
+
+// StorageVolumeConfig represents the fields of a LXD storage volume.
+type StorageVolume struct {
+	VolumeType       string   `json:"type" yaml:"type"`
+	VolumeUsedBy     []string `json:"used_by" yaml:"used_by"`
+	StorageVolumePut `yaml:",inline"`
```

Comment:
Inheritance should be on the first line of the struct with blank separator line afterwards.